CHANGELOG.md

```diff
@@ -3,6 +3,14 @@
 
 ## New features
 
+- Export functions can have `reg ptr` as arguments and results.
+  The compiler assumes that writable `reg ptr` are disjoint from the other
+  `reg ptr` arguments and from the global data. This is the responsibility of
+  the caller to ensure that this holds.
+  For now, writable `reg ptr` must come first in the list of arguments and be
+  returned first and in the same order in the list of results.
+  ([PR #707](https://github.com/jasmin-lang/jasmin/pull/707)).
```

Comment:
```suggestion
- Export functions can have arrays as arguments and results.
  The compiler assumes that writable `ptr` are disjoint from the other
  `ptr` arguments and from the global data. This is the responsibility of
  the caller to ensure that this holds.
  For now, writable `ptr` must come first in the list of arguments and be
  returned first and in the same order in the list of results.
  ([PR #707](https://github.com/jasmin-lang/jasmin/pull/707)).
```

I think that the wording “reg ptr” is wrong and misleading.

Reply:
The answer was hidden in compiler_proof.v
```
(** Export functions (entry points) have restrictions on ptr arguments and returns.
    Namely, the reg mut ptr should be the first arguments and returned at the same
    positions in the return values.
    This is not a constraint coming from the implementation, it is just meant to give
    a readable correctness theorem. *)
```